numpy/core/src/umath/_rational_tests.c.src

```diff
@@ -727,13 +727,13 @@ npyrational_setitem(PyObject* item, void* data, void* arr) {
         r = ((PyRational*)item)->r;
     }
     else {
-        long n = PyInt_AsLong(item);
+        long n = PyLong_AsLong(item);
         PyObject* y;
         int eq;
         if (error_converting(n)) {
             return -1;
         }
-        y = PyInt_FromLong(n);
+        y = PyLong_FromLong(n);
```

Comment:
I think these are both wrong and should be long long. I suppose that's a follow up of "review all callers of `PyLong_FromLong`"

Reply:
Agree, have made that change.